src/content/en/fundamentals/getting-started/primers/media-source-extensions.md

```diff
@@ -0,0 +1,299 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: 
+
+{# wf_published_on: 2017-02-08 #}
+{# wf_updated_on: 2017-02-08 #}
+
+# Media Source Extensions {: .page-title }
+
+{% include "web/_shared/contributors/josephmedley.html" %}
+
+Media Source Extensions (MSE) is a JavaScript API that lets you build streams
+for playback from segments of audio or video. Although not covered in this
+article, understanding MSE is needed if you want to embed videos in your site
+that does such things as:
+
++  Adaptive streaming, which is another way of saying adapting to device
+   capabilities and network conditions
++  Adaptive splicing
++  Time shifting
++  Control performance and download size
+
+You can almost think of it as a chain. Between the downloaded file and the media
+elements are several layers.
+
++  An `<audio>` or `<video>` element to play the media.
++  A `MediaSource` instance with a `SourceBuffer` to feed the media element.
++  A fetch or XHR call to retrieve media data in a Response object.
++  A call to `Response.arrayBuffer()` to feed `MediaSource.SourceBuffer`.
+
+This is illustrated below. 
+
+![Basic MSE data flow](imgs/basic-mse-flow.png)
+
+In practice, the chain looks like this:
+
+    var vidElement = document.querySelector('video');
+    
+    if (window.MediaSource) {
+      var mediaSource = new MediaSource();
+      vidElement.src = URL.createObjectURL(mediaSource);
+      mediaSource.addEventListener('sourceopen', sourceOpen);
+    } else {
+      console.log("The Media Source Extensions API is not supported.")
+    }
+    
+    function sourceOpen(e) {
+      var mime = 'video/webm; codecs="opus, vp9"';
+      var mediaSource = this;
+      var sourceBuffer = mediaSource.addSourceBuffer(mime);
+      var video = 'droid.webm';
```

Comment:
I think you should explain why you're only using a single video source in this article. MSE only makes sense in practice for scenarios with multiple segments of video. An indented note would probably be enough.